guppylang/decorator.py

```diff
@@ -418,13 +410,15 @@ def compile_function(
             mono_args = tuple(None for _ in checked_def.params)
 
         compiled_def = ENGINE.compiled[obj.id, mono_args]
-        assert isinstance(compiled_def, CompiledFunctionDef | CompiledPytketDef)
```

Comment:
We were missing a lot of definition kinds in this assert. The new `CompiledHugrNodeDef` ABC makes this check more uniform